### DISCUSSION TITLE
Whodunnit: brainstorming changes to the "find the leak" mission

### DISCUSSION THREAD

**Discussion by I-am-Erk** [CleverRaven/Cataclysm-DDA#52148] [Ideas]
I would like to change the old guard "find the leak" mission to be quite an involved 'whodunnit' style mission that takes you around the center, finding clues and asking questions from everyone in the center until you gain enough information to accuse the informant, either to their face or to the old guard rep. Later, this can also be a chance to keep the informant's secret and either blackmail them or get them to connect you to the Hell's Raiders.

This is a complex kind of quest and I'd love ideas for some of the clues and how to uncover them.

Here's what I have so far:

## Setup
The old guard rep tells you they believe they have a leak in the base, same as current. They have only a suspicion so far. Would you be able to investigate?

In prep for this quest, two guards in the center will be given non-random names and genders so that they can be referred to by name. One of those guards will be the informant and the other will be someone they've set up as a possible patsy. A few other guards could be given static names to allow you to accuse a few other randos.

### Asking around
Several things can be learned by asking around. Some of these things require you to have learned a bit more about the case first.
- If you have progressed in the Relay subpart, you can find out from various members of the refugee center whether or not they know anything about electronics. The informant will admit they know nothing, thinking this reduces their suspicion, and the patsy will admit to being pretty good at it and offer to train you for a price.

### The Relay
*In order to get transmissions this far, the informant needs to have a way to pass messages at a distance. They are sending short range signals to some kind of booster relay and have set something up on a nearby radio tower.*
**First stage**
- "relay_range": Can be attained from the old guard rep if player has a high electronics or intelligence, or from at least 2 other people on base. Someone observes the range problem, how is the informant getting signals out? It could be through letters or parcels but if it's radio they either need something long range, or a relay. Given the quick transmission of information, radio seems more likely.
- "relay_observed": Only attainable once "relay_range" attained. One caravan guard has noticed a strange jury-rigged looking package on a nearby radio tower when travelling out from the base. If the teamster is available, they'll give you this clue, otherwise one of the guards can do it. Grants access to second stage.
- "relay_parts_1": Only attainable once "relay_range" attained. Jenny Forcette can talk to you about what parts would be needed to build a relay.
- "relay_parts_2": Follow up to "relay_parts_1", the free merchant broker can confirm after a quick check that several of the necessary parts Jenny described are missing from their inventory.

**Second stage**
"relay_observed" assigns a mission to go check out the radio tower and updates its mapgen to include some suspicious stuff.
Exploring the tower, you find a relay and can either:
- study it and memorize its construction (requires electronics + intelligence or a memory-related cbm or mutation)
- take notes on how it is built (requires paper, electronics)
- take a picture of it (requires a camera of some kind)
- wish you had some way to photograph or draw details so you could remember them later
The first three options are considered equivalent.
You can choose to destroy the relay. This will turn increment a variable, "informant_wary", which tracks how concerned the informant is about your progress.

**Third stage**
You can report your findings about the relay back to the old guard rep, and if you have "relay_parts_1" or "_2" and some kind of record of things, you can also share your observations with Jenny or the broker.
- Jenny will observe that the device is crudely made by someone with only basic knowledge of electronics. Gives clue "informant_not_handy".
- The broker will note that not all the missing parts seem to have been used in that design, and tracking down the extra parts might be helpful. Very little gets thrown out in the center, they suggest you check the guards' lockers. You can find the parts in one of the guards' lockers, belonging to the patsy guard. This gives clue "patsy_has_parts". You can point this out to the old guard rep, but they would like more evidence before acting on it.

### The Motive
*The informant has to have some reason for risking themselves like this.  Who in the center would have such a grudge against the merchants? It turns out, quite a few people feel pretty angry about how the situation in the Back Bay was handled, and the fact that a lot of the people who could have done something and didn't are now holding positions of power in the Free Merchants.*
- Asking around: Nearly all the refugees have a bit of a chip on their shoulders about the Back Bay events.
- This quest will require writing up a power structure for the Free Merchants.
- ties into "the coverup"

### The Coverup
*As you dig deeper, you get more information on what happened in the dreaded 'back bay' to lead to all those deaths. This may lead to a further uprooting of some deeper problems among the free merchants, helping solve this case but also leading to some new drama.*

## Finishing the mission
In order to make an accusation you need one major and two minor pieces of evidence, or two major pieces of evidence.
**Major**:
"patsy_has_parts" - if you are accusing the patsy

**Minor**:
"informant_not_handy" in conjunction with clue from interrogating the refugees to confirm it applies to the informant
"patsy_electronics_skilled" can be gained by talking to the patsy, is considered a minor clue if you have "patsy_has_parts" but if you have "informant_not_handy" it is invalidated.

**Comment by PatrikLundell** [comment]
I'd probably want to add an information trap of some kind, i.e. device a way to feed different info to the two (or more) suspects and see which info the raiders acted on. You'd probably not want to sacrifice a caravan for this, but you might be willing to risk a supply stash. If so, you could have an option for the player to set up an ambush by the stash (if they're fed to suspects at different times), with the representative making it very clear that this is very risky and not necessary.

I'm not that keen on the self reported skills of the suspects as important, as it's easy to lie about that (and probably to fake incompetence if aware of it being a test). I'd probably want to examine examples of repairs they've had to do in the field (if guards occasionally accompany caravans that may happen).

I like the idea of coverup dirt being uncovered pointing towards the current downstairs power structure (separate thread), and wouldn't mind seeing indications of the elimination of (more competent) competition by an unscrupulous individual.

"Finishing the mission" above mentions doesn't currently add up. There aren't two major pieces of evidence, and only two (at most) minor ones. It seems this bit is unfinished.

I'd like to have two different scenario for the ending, one where you report the evidence and call it a day, and one where you participate in the arrest. In the latter case I'd like to see a peaceful arrest (on successful conversation outcomes) and a violent one (on a failure), possibly with you having to push further for an incorrect arrest to become violent.

Edit: Some additional thoughts.
What about "updating" the relay such that it also relays the transmission onto another frequency the representative monitors and records? If you're able to record the voice of the mole you should be able to use that to catch it (especially if the mole is of a different gender to the designated scapegoat). A simpler alternative for those with some, but lesser, electronics skills might be to attack a recorder to the relay itself.
In fact, if you're able to determine the frequency the relay transmits on, you don't need to modify it: just monitor that frequency. I'm not sure if you can have both modification and frequency determination as different options base on the skills available, or it they both would boil down to Electronics.
